src/stan/math/prim/mat/functor/finite_diff_gradient.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef STAN__MATH__PRIM__MAT__FUNCTOR__FINITE_DIFF_GRADIENT_HPP
+#define STAN__MATH__PRIM__MAT__FUNCTOR__FINITE_DIFF_GRADIENT_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+
+namespace stan {
+  
+  namespace math {
+
+    /**
+     * Calculate the value and the gradient of the specified function
+     * at the specified argument using finite difference.  
+     *
+     * <p>The functor must implement 
+     * 
+     * <code>
+     * double
+     * operator()(const
+     * Eigen::Matrix<double,Eigen::Dynamic,1>&)
+     * </code>
+     *
+     * Error should be on order of epsilon ^ 6
+     * 
```

Comment:
and the number of function calls to calculate the finite diff gradient using this method